Makefile

```
@@ -258,11 +258,12 @@ vulncheck:
 	@go run golang.org/x/vuln/cmd/govulncheck@latest ./...
 .PHONY: vulncheck
 
-#? gofumpt-pre-commit: Create gofumpt pre-commit hook
-gofumpt-pre-commit:
-	@echo "--> Creating gofumpt pre-commit hook"
-	@mkdir -p .git/hooks
-	@cp scripts/gofumpt.sh .git/hooks/pre-commit
+#? setup-pre-commit: Create git pre-commit hook for gofumpt, golangci-lint
+setup-pre-commit:
+	@cp .git/hooks/pre-commit .git/hooks/pre-commit.bak 2>/dev/null || true
+	@echo "Installing pre-commit hook..."
+	@ln -sf ../../scripts/hooks/pre-commit.sh .git/hooks/pre-commit
```

```suggestion
	@ln -sf scripts/hooks/pre-commit.sh .git/hooks/pre-commit
```

we're executing this from root dir, no?

---

Yes, we are in root dir. But we should still use `../../scripts/hooks/pre-commit.sh`. I have tested this, `scripts/hooks/pre-commit` couldn't work well.

I think the path is refer to the target to be linked, not the dir that command execute